lib/maze/browser_stack_utils.rb

```diff
@@ -21,19 +21,26 @@ def upload_app(username, access_key, app)
           request.basic_auth(username, access_key)
           request.set_form({ 'file' => File.new(expanded_app, 'rb') }, 'multipart/form-data')
 
-          res = Net::HTTP.start(uri.hostname, uri.port, use_ssl: true) do |http|
-            http.request(request)
-          end
+          upload_tries = 0
 
-          begin
-            body = res.body
-            response = JSON.parse body
-            raise "Upload failed due to error: #{response['error']}" if response.include?('error')
-            raise "Upload failed, response did not include and app_url: #{res}" unless response.include?('app_url')
-          rescue JSON::ParserError
-            raise "Error: expected JSON response, received: #{body}"
-          end
+          while upload_tries < 3
+            res = Net::HTTP.start(uri.hostname, uri.port, use_ssl: true) do |http|
+              http.request(request)
+            end
 
+            begin
+              body = res.body
+              response = JSON.parse body
+              raise "Upload failed due to error: #{response['error']}" if response.include?('error')
+              raise "Upload failed, response did not include and app_url: #{res}" unless response.include?('app_url')
+            rescue JSON::ParserError
+              raise "Error: expected JSON response, received: #{body}"
+            end
+
+            upload_tries = 3 if !response.include?('error') 
+            upload_tries = tries.next unless !response.include?('error')
```

Comment:
I think you could achieve the same aim with less code by:
- Adding a `break` immediately before the `rescue` line (i.e. assume success if it gets past the previous lines)
- Remove these lines in preference to simply `upload_tries += 1`